include/vcpkg/base/stringview.h

```diff
@@ -86,3 +88,45 @@ namespace vcpkg
 
 VCPKG_FORMAT_AS(vcpkg::ZStringView, vcpkg::StringView);
 VCPKG_FORMAT_AS(vcpkg::StringLiteral, vcpkg::StringView);
+
+template<::size_t N>
+struct StringArray : public std::array<char, N>
+{
+    constexpr StringArray() noexcept : std::array<char, N>() { }
+    constexpr StringArray(const char (&str)[N]) noexcept : std::array<char, N>()
+    {
+        for (size_t i = 0; i < N; i++)
+        {
+            this->at(i) = str[i];
+        }
+    }
+
+    constexpr ::size_t length() const noexcept { return this->size() - 1; }
```

Comment:
```suggestion
    constexpr ::size_t size() const noexcept { return this->size() - 1; }
```

The STL convention for containers is that they have `size`, not `length`.

Reply:
Yes, but there is `std::string.length()`

Reply:
Right, but that's a legacy holdover from before `std::string` gained an STL container interface.